util/fibers/listener_interface.cc

```diff
@@ -193,10 +196,12 @@ void ListenerInterface::RunAcceptLoop() {
       if (guard->hook_.is_linked()) {
         ++iter;
       } else {
+        VLOG(1) << "Iterator reset for pb with thread_id " << pb->thread_id();
```

Comment:
This `else` loop could lead us to a never ending `while()` loop. Let's see if the following logs provide us with more info. Also Roman, if you have anything to add here I am all eyes :eyes: